catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -150,20 +140,128 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
+                            registryId)))
+                            .get(0));
+        } catch (FederationAdminException e) {
+            String message = "Error getting Metacard.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        try {
+            html = registryReportBuilder.getHTMLFromMetacard(metacard);
         } catch (IOException e) {
-            LOGGER.error("Error compiling and applying report template.");
+            String message = "Error in compiling and applying report template.";
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
         }
 
         return Response.ok(html)
                 .build();
     }
 
+    @Path("/{registryId}/report/{section}")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewSectionInfoHtml(@PathParam("registryId") final String registryId,
+            @QueryParam("sourceId") final List<String> sourceIds,
+            @PathParam("section") final String section) {
+        String html = "";
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            Set<String> sourceIdsSet = new HashSet<>();
+            if (CollectionUtils.isNotEmpty(sourceIds)) {
+                sourceIdsSet.addAll(sourceIds);
+            }
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIdsSet);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For registry id: '{}', optional sources: {}",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(message)
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For registry id: '{}', optional source ids: {}.",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(message)
+                    .build();
+        }
+
+        if(!isValidSection(section)) {
+            String message = section + " is an unknown section name";
```

Comment:
Thank you for the feedback, I will be modifying the code to include all of the above!